packages/connect-core/src/utils/metadata.ts

```diff
@@ -19,22 +20,26 @@ export function parseMetadata(name: string, metadata: string): any {
   }
 }
 
+const metadataCacheStore = createCacheStore<object>(10)
+
 export async function fetchMetadata(
   fileName: string,
   contentUri: string
 ): Promise<any> {
-  const contentHashRegEx = contentUri.match(/ipfs:(.*)/)
-  if (contentHashRegEx) {
-    const url = `${DEFAULT_IPFS_GATEWAY}/ipfs/${contentHashRegEx[1]}/${fileName}`
-    let metadata
-    try {
-      metadata = await ethersUtils.fetchJson(url)
-    } catch (error) {
-      throw new ErrorConnection(`Couldn’t fetch ${url}, failed with error.`)
-    }
-    return metadata
+  const contentHash = contentUri.match(/ipfs:(.*)/)?.[1]
+  if (!contentHash) {
+    return {}
+  }
+
+  const url = `${DEFAULT_IPFS_GATEWAY}/ipfs/${contentHash}/${fileName}`
+  try {
+    // await before returning to catch any error in the callback
+    return await metadataCacheStore.get(url, () =>
+      fetch(url).then((res) => res.json())
+    )
+  } catch (error) {
+    throw new ErrorConnection(`Couldn’t fetch ${url}, failed with error.`)
```

Comment:
Here we don't want to provide further details on the error other than it is a Connection related error?

Reply:
Are you thinking of having a different types of error for the actual fetching, the JSON parsing, and the cache store?

Reply:
No, I think the Connection type is fine. My question was to have a better understanding of when we should display the `error` message to give further details. And when we can mute the message.

Reply:
I made it throw two different types of error: https://github.com/aragon/connect/pull/296/commits/444113d59bfd858cc78251a1562340cb6d646723